persistence/db.go

```diff
@@ -65,31 +68,57 @@ func (pg *PostgresContext) ResetContext() error {
 	return nil
 }
 
-// TECHDEBT: Implement proper connection pooling
-func connectToDatabase(postgresUrl string, schema string) (*pgx.Conn, error) {
+func connectToDatabase(cfg *configs.PersistenceConfig) (*pgx.Conn, error) {
 	ctx := context.TODO()
 
-	conn, err := pgx.Connect(context.Background(), postgresUrl)
+	config, err := pgxpool.ParseConfig(cfg.GetPostgresUrl())
+	if err != nil {
+		return nil, fmt.Errorf("unable to create database config: %v", err)
+	}
+	maxConnLifetime, err := time.ParseDuration(cfg.GetMaxConnLifetime())
+	if err == nil {
+		config.MaxConnLifetime = maxConnLifetime
+	} else {
+		log.Printf("unable to set max connection lifetime: %v", err)
```

Comment:
@Prajjawalk Do you think just logging it (rather than returning an error) is more appropriate?

I was thinking that if we return `nil, err`, it'll inform the operator their configs are misconfigured and fix it. Not a lot of people look at logs :)

Reply:
If we return an error instead of logging then it may break the execution flow throwing a runtime error, I thought if configs are misconfigured then the error may not be that critical to break the flow as we can resort to default while informing user through logs, what do you think @Olshansk ?

Reply:
I don't think there's a right answer here as I can definitely argue both sides of it. I'd say it's probably a function of what were the worse situation a developer has experienced in the past:

1. Changing configs and spending a lot of time figuring out why they weren't being applied.
2. Misconfiguring configs and having everything crash.

There's a case to be made for each :)